core/src/array/array_sorted_read_state.cc

```diff
@@ -2437,30 +2088,57 @@ Status ArraySortedReadState::read_dense_sorted_col() {
     return array_->read_default(
         copy_state_.buffers_, copy_state_.buffer_sizes_);
 
-  // Iterate over each tile slab
+  // Handle resume copy
+  if (resume_copy_) {
```

Comment:
You it doesn't look pretty but you definitely do this, just google search function pointer to member function.

Also C++11 has `std::function` which makes this a bit cleaner.